keps/sig-cli/kustomize-annotate-generated.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Kustomize Annotate Generated Secrets and ConfigMaps
+authors:
+ - "@pwittrock"
+owning-sig: sig-cli
+participating-sigs:
+ - sig-cli
+reviewers:
+ - "@janetkuo"
+ - "@anguslees"
+approvers:
+ - "@monopole"
+editors:
+ - "@pwittrock"
+creation-date: 2019-03-01
+last-updated: 2019-03-01
+status: provisional
+see-also:
+replaces:
+superseded-by:
+ - n/a
+---
+
+# Kustomize Annotate Generated Secrets and ConfigMaps
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+- Annotate generated Secrets and ConfigMaps with Workloads the reference them
+- Annotated Workloads with the Secrets and ConfigMaps they reference
+- Workload annotations are retained after the Workloads stop referencing the Secrets and ConfigMaps
+- Secret and ConfigMap annotations are retained after the Workloads stop referencing them
+
+## Motivation
+
+- Kustomize generates Secret and ConfigMap Resources which are referenced by Workloads
+  - When the data for the Secrets or ConfigMaps change, new Secret or ConfigMap Resources are generated
+  - Kustomize updates the Workloads to refer to the new Resources, and stop referring to the old Resources
+  - References to the old Secret and ConfigMap Resources are leaked when new Resources are generated
+  - The information that the Resources were generated by Kustomize rather than explicitly defined by
+    users is lost
+
+### Goals
+
+- Provide provenance for Resources generated by Kustomize
+  - Retain a record of Secret and ConfigMap Resources generated by Kustomize.
+  - Retain a record of which Workloads each generated Secret and ConfigMap was consumed by.
+  - Retain a record of which generated Secret and ConfigMap Resources each Workload consumed.
+- Allow other tools to identify generated Secrets and ConfigMaps that are no longer in use as
```

Comment:
Can we include some pseudo-code in here, describing the intended operations using these annotations?  In particular, I think we should move some of them to labels (selectors) instead - and this will become clearer when we look at how these are going to be used in practice.